UI/ChatGPT/src/ChatGPT/Services/ChatService.cs

```diff
@@ -43,56 +64,61 @@ public async IAsyncEnumerable<ChatResponse> AskAsStream(ChatRequest chatRequest,
 		var response = new ChatResponse();
 		var content = new StringBuilder();
 
-		IAsyncEnumerator<ChatCompletionCreateResponse>? responseStream = default;
-		while (!response.IsError)
+		AsyncResultCollection<StreamingChatCompletionUpdate> result = _client.CompleteChatStreamingAsync(request);
+		var enumerator = result.GetAsyncEnumerator(ct);
+
+		try
 		{
-			try
+			while (await enumerator.MoveNextAsync())
 			{
-				responseStream ??= _client.CreateCompletionAsStream(request).GetAsyncEnumerator(ct);
-				if (await responseStream.MoveNextAsync())
+				var update = enumerator.Current;
+
+				try
 				{
-					if (responseStream.Current.Successful)
+					foreach (ChatMessageContentPart updatePart in update.ContentUpdate)
 					{
-						foreach (var choice in responseStream.Current.Choices)
+						try
 						{
-							content.Append(choice.Message.Content);
+							content.Append(updatePart.Text);
+							response = response with { Message = content.ToString() };
+						}
+						catch (Exception)
+						{
+							response = response with { Message = "Something went wrong.", IsError = true };
 						}
-						response = response with { Message = content.ToString() };
-					}
-					else
-					{
-						response = response with { Message = responseStream.Current.Error?.Message, IsError = true };
 					}
 				}
-				else
+				catch (Exception)
 				{
-					yield break;
+					response = response with { Message = "Something went wrong.", IsError = true };
 				}
-			}
-			catch (Exception)
-			{
-				response = response with { IsError = true };
-			}
 
-			yield return response;
+				yield return response;
+			}
+		}
+		finally
+		{
+			await enumerator.DisposeAsync();
 		}
 	}
 
-	private ChatCompletionCreateRequest ToCompletionRequest(ChatRequest request)
+	private IEnumerable<ChatMessage> ToCompletionRequest(ChatRequest request)
 	{
 		var history = request.History;
 		var requestMessages = new List<ChatMessage>(history.Count + 1)
 		{
-			ChatMessage.FromSystem(systemPrompt)
+			ChatMessage.CreateSystemMessage(systemPrompt)
```

Comment:
There is something weird going on here.

The app is working well on Desktop and Windows. But on Android I'm getting this exception on this line:

https://github.com/unoplatform/Uno.Samples/blob/6853e3c4e70cc1a559c1295fb9959789b2dfd5c4/UI/ChatGPT/src/ChatGPT/Services/ChatService.cs#L55

```
HTTP 400 (invalid_request_error: )
Parameter: messages.[0].content

Invalid value for 'content': expected a string, got null.
```

See below the history (list of messages) being sent ([this line](https://github.com/unoplatform/Uno.Samples/blob/6853e3c4e70cc1a559c1295fb9959789b2dfd5c4/UI/ChatGPT/src/ChatGPT/Services/ChatService.cs#L26)) :

### Desktop
![chatgpt-desktop](https://github.com/unoplatform/Uno.Samples/assets/114886335/76021a62-7656-4b8d-b5fe-5b296c633737)


### Android
![chatgpt-android](https://github.com/unoplatform/Uno.Samples/assets/114886335/dc9a62df-3217-4e42-bd7b-4d03a20682d4)

cc @dr1rrb @agneszitte

Reply:
Isnt this because of CORS somehow for WASM?

Reply:
The issue we are facing on Android for tracking - https://github.com/openai/openai-dotnet/issues/30

Reply:
It's now a dotnet bug: https://github.com/dotnet/runtime/issues/103365

Reply:
Fixed on their latest package.